### ISSUE TITLE
Problems with desktop environments copying .desktop files

### ISSUE THREAD

**Issue by ambrop72** [NixOS/nixpkgs#11471]
Some desktop environments will copy a .desktop file to a local directory in the user "transfers" it somewhere, for example to the desktop, to the panel or to the favorites. This happens with both KDE4 and Xfce. While the shortcuts which refer to the program by name only will be okay, but those which specify the absolute path to the executable will still refer to the old application or would be broken if a GC deleted the old application.

Ideally the system shortcuts in this way would establish some kind of reference rather than a literal copy of the .desktop file.

**Comment by mogorman** [comment]
I don't think this is fixable unless we changed .desktop files to link to current-system instead of their full nix paths

**Comment by vcunat** [comment]
Are the absolute paths in *.desktop good for anything? What if we had a hook that converts them to name-only?

**Comment by Profpatsch** [comment]
triage: what’s the status?

**Comment by matthewbauer** [comment]
> Are the absolute paths in *.desktop good for anything? What if we had a hook that converts them to name-only?

We now have a PR that goes the other way! It converts the name-only ones to absolute path: https://github.com/NixOS/nixpkgs/pull/68035
